src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -128,6 +136,8 @@ public override Task<Compilation> TransformCompilationAsync(Compilation oldCompi
                 {
                     return Task.FromResult(oldCompilation.WithOptions(_options));
                 }
+
+                public override bool CanApplyToCompilationWithGeneratedDocuments => true;
```

Comment:
None of this is used to prevent a re-run of the generators -- we'll still run them either way. It's only to optimize for the reuse of the prior compilation if we are able to do so. In this case, the "true" here doesn't mean "don't run generators" but "if the output is the same, we can reuse the prior compilation". For example, we can't do that for a parse option change since we have to re-parse all trees no matter what.

Reply:
> if the output is the same, we can reuse the prior compilation

`bool CanReusePriorCompilationIfOutputIsTheSame { get; }` ?

Reply:
i admit the current name is a bit confusing.

Reply:
Trying CanUpdateCompilationWithStaleGeneratedTreesIfGeneratorsGiveSameOutput which I admit is terribly wordy but I think combines:

1. Making it clear it's only used if the generators give the same output (@sharwell's original question). "IfOutputIsTheSame" might immediately make somebody ask "what 'output'"?
2. A key which the translation action this method is on isn't doing any reuse analysis, the question here is whether the action can do the update on a compilation safely.